### PR TITLE
fix #53 wrong response code on overzoom requests

### LINKED ISSUE #53: Support overzoom
I'm displaying vector tiles from a OSM mbtiles country extract using MapboxGL that has data up to zoom level 14. When I overzoom past the supported zoom level then I just see a blank map. Tessera is returning empty tiles rather than a '404 tile not found'. 

It would be great if Tessera could respond with a 404 instead because this would allow you to zoom all the way in (albeit with reduced resolution).

### PR DESCRIPTION
**Pull request by ramunasd**
this also introduces functional tests, based on mocha

### CODE REVIEW

**Comment by mojodna** [comment]
I'm a little backed up, so it may be a few days before I can fully review this.

Can you summarize the changes, specifically how this changes behavior? Thanks!

**Comment by ramunasd** [comment]
Well, the main idea is to add additional argument to getTile callback function. This argument is http status code, which must be returned. So now http status code is determined in one place, during request validation and getTile callback just sends everything as it gets :)

**Comment by ramunasd** [comment]
Hi @mojodna would be nice to get Your feedback ;)
